File: artisanat-frontend/src/app/core/dtos/response/order-response-dto.ts
import {Status} from "../../enums/status.enum";
import {Product} from "../../models/product.model";
import {Customer} from "../../models/customer.model";
import {SubOrderResponseDto} from "./sub-order-response-dto";

export interface OrderResponseDto {
  id: number;
  quantity: number;
  orderDate: string;
  location: string;
  status: string;
  totalAmount: number;
  customer: Customer;
  subOrders: SubOrderResponseDto[];
}

File: artisanat-backend/src/main/java/com/artisanat_backend/dto/response/OrderResponseDto.java
package com.artisanat_backend.dto.response;

import com.artisanat_backend.enums.Status;
import com.artisanat_backend.model.Customer;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class OrderResponseDto {
    private Long id;
    private int quantity;
    private String orderDate;
    private String location;
    private Status status;
    private double totalAmount;
    private Customer customer;
    private List<SubOrderResponseDto> subOrders;
}
